main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,506 @@
+  or `process.argv` for the argument array.
+* `URL` and `URLSearchParams`: Are not available.
+* `WebAssembly`: Is not available.
+* `TextEncoder` and `TextDecoder`: Are not available.
+
+Some names look like globals, but are really part of the module-defining tools: imports, exports, and 
+metadata. Modules start as files on disk, but then are bundled together into an archive before being 
+loaded into a vat. The bundling tool uses several standard functions to locate other modules that must 
+be included. These are not a part of SES, but are allowed in module source code, and are translated or 
+removed before execution. 
+
+- `import` and `export` syntax are allowed in ESM-style modules ( preferred over CommonJS). These are 
+  not globals per se, but top-level syntax that defines the module graph.
+- `require`, `module`, `module.exports`, and `exports` are allowed in CommonJS-style modules, and 
+  should work as expected. However, new code should be written as ESM modules. They are either 
+  consumed by the bundling process, provided (in some form) by the execution environment, or 
+  otherwise rewritten to work sensibly
+- `__dirname` and `__filename` are not provided
+- The dynamic import expression (`await import('name')`) is currently prohibited in vat code, 
+  but a future SES implementation may allow it.
+
+Node.js has a [large collection](https://nodejs.org/dist/latest-v14.x/docs/api/) of "built-in 
+modules", such as `http` and `crypto`. Some are clearly platform-specific (e.g. `v8`), while 
+others are not so obvious (`stream`). All are accessed by importing a module (`const v8 = require('v8')` 
+in CommonJS modules, or `import v8 from 'v8'` in ESM modules). These modules are built out of 
+native code (C++), not plain JavaScript.
+
+None of these built-in modules are available to SES code. `require` or `import` can be used on 
+pure JavaScript modules, but not on modules including native code. To exercise authority from 
+a built-in module, you have to write a *device* with an endowment with the built-in module's 
+functions, then send messages to the device.
+
+Browser environments also have a huge list of [other features](https://developer.mozilla.org/en-US/docs/Web/API) 
+presented as names in the global scope (some also added to Node.js). None are available in a 
+SES environment. The most surprising removals include `atob`, `TextEncoder`, and `URL`.
+
+`debugger` is a first-class JavaScript statement, and behaves as expected.
+
+## What does SES add to standard Javascript
+
+- `console` is available to help with debugging. Since all JavaScript implementations
+  add it, you may be surprised it’s not in the official spec. So leaving it out would cause 
+  too much confusion. Note that `console log`’s exact behavior is up to the host program; display 
+  to the operator is not guaranteed. Also, it is difficult at best to write to `process.stdout` and
+  so that should not be done.
```

Comment:
```suggestion
  to the operator is not guaranteed. Use the console for debug information only. The console is not obliged to write to the POSIX standard output and is a poor substitute for expressly threading that capability into a compartment.
```

Reply:
I've put the whole thing in for now, but do our developers need to know the info either in all of the second sentence (The console is not...) or the second part (is a poor substitute)? It feels a little deep for users as opposed to internal devs?

Reply:
We can omit “is a poor substitute” and beyond.